Patch: 74_admin_non-ascii_emails.patch
Author: Lionel Elie Mamane <lionel@mamane.lu>
Don't assume emails are ASCII (Closes: #315358)
Index: b/Mailman/Cgi/admin.py
===================================================================
--- a/Mailman/Cgi/admin.py	2010-05-24 17:50:44.000000000 +0200
+++ b/Mailman/Cgi/admin.py	2010-05-24 17:51:14.000000000 +0200
@@ -867,7 +867,12 @@
     chunksz = mlist.admin_member_chunksize
     # The email addresses had /better/ be ASCII, but might be encoded in the
     # database as Unicodes.
-    all = [_m.encode() for _m in mlist.getMembers()]
+    all = []
+    for _m in mlist.getMembers():
+        try:
+            all.append( _m.encode() )
+        except:
+            all.append( _m )
     all.sort(lambda x, y: cmp(x.lower(), y.lower()))
     # See if the query has a regular expression
     regexp = cgidata.getvalue('findmember', '').strip()
